server/src/main/java/com/linecorp/centraldogma/server/internal/mirror/GitMirror.java

```diff
@@ -347,7 +347,12 @@ private Git openGit(File workDir) throws Exception {
                 CONSECUTIVE_UNDERSCORES.matcher(DISALLOWED_CHARS.matcher(
                         remoteRepoUri().toASCIIString()).replaceAll("_")).replaceAll("_"));
 
-        final GitWithAuth git = new GitWithAuth(this, repoDir);
+        GitWithAuth git;
+        try {
+            git = new GitWithAuth6(this, repoDir);
+        } catch (Exception e) {
+            git = new GitWithAuth5(this, repoDir);
+        }
```

Comment:
You need to use reflection here because `:server` can't depend on `:server-jgit6`, i.e.

```java
try {
  Class type6 = Class.forName(...);
  Constructor constructor6 = type6.getConstructor(GitMirror.class, File.class);
  git = constructor6.newInstance(this, repoDir);
} catch (Exception e) {
  git = new GitWithAuth5(...);
}
```

Reply:
updated, thanks